etc/spec/SPECIFICATION.md

```diff
@@ -117,12 +118,28 @@ is verified as follows:
  - The symbols . , / ? ~ ! @ # $ % ^ & * ( ) - _ = + [ ] { } \ | ' " ` ; :
 3. The username must not be reserved.
 
-The username is also validated for form, but not for reservation, as the
-username is being typed.
+A username is reserved if and only if someone is registered with that username
+or someone is using it at that moment (even if that username is not registered).
 
-All usernames are not reserved by default. If the username is valid, the
-username is reserved and the user is taken to the Chat screen. The username is
-then associated with that user. The user's credentials are then stored on the
+While the user type a username it is valid, ting is searching for that username
+to check if it is registered.
+
+1. If the username is registered then a new form appears which asks from the
+   user to enter the password that is associated with the username. If the password
+   is correct then user is taken to the Chat screen by clicking the button "Μπες".
```

Comment:
So how will this work? If a new form appears while I am typing half of my username, won't it interrupt my typing?